app/main.py

```diff
@@ -2,5 +1,16 @@
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    result = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        if args not in result:
```

Comment:
Using `args` as a key in the dictionary assumes that all arguments are hashable. If the function accepts non-hashable arguments (like lists or dictionaries), this will raise a `TypeError`. Consider handling such cases or documenting that the decorator only works with hashable arguments.